maint/scripts/run_local_ci_test.sh

```
@@ -1,22 +1,197 @@
+  # Non-CUDA: do not auto-detect devices; do not load CUDA plugin
+  NWORKERS_NONCUDA=${PYTEST_XDIST_WORKERS:-4}
+  if ! [[ "$NWORKERS_NONCUDA" =~ ^[0-9]+$ ]] || [[ "$NWORKERS_NONCUDA" -le 0 ]]; then
+    NWORKERS_NONCUDA=4
+  fi
+  echo "[INFO] DEVICE=$DEVICE; running without CUDA plugin. Workers: $NWORKERS_NONCUDA."
+  PYTEST_ARGS_DEVICE=(-n "$NWORKERS_NONCUDA")
+fi
 
-# Run pytest in parallel (4 workers) for all tests in the examples directory
+# Run pytest in parallel for all tests in the examples directory
 cd examples
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)